traits/trait_types.py

```diff
@@ -2774,6 +2774,178 @@ def items_event(self):
 AdaptMap = {"no": 0, "yes": 1, "default": 2}
 
 
+class Map(TraitType):
+    """ Checks that the value assigned to a trait attribute is a key of a
+        specified dictionary, and also assigns the dictionary value
+        corresponding to that key to a *shadow* attribute.
+
+        A trait attribute of type Map is called a *mapped* trait
+        attribute. In practice, this means that the resulting object actually
+        contains two attributes: one whose value is a key of the Map
+        dictionary, and the other whose value is the corresponding value of the
+        Map dictionary. The name of the shadow attribute is simply the base
+        attribute name with an underscore ('_') appended. Mapped trait
+        attributes can be used to allow a variety of user-friendly input values
+        to be mapped to a set of internal, program-friendly values.
+
+        Example
+        -------
+
+        The following example defines a ``Person`` class::
+
+            >>>class Person(HasTraits):
+            ...    married = Map({'yes': 1, 'no': 0 })
+            >>>
+            >>>bob = Person()
+            >>>print bob.married
+            yes
+            >>>print bob.married_
+            1
```

Comment:
This example is not consistent with the code.  Also, please use `print()` (Python 3).